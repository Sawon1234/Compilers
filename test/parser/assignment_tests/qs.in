fun quicksort(l : List) : List
{
	m := l.first();
	l.pop();
	a := newList();
	b := newList();
	for(i in 1..(l.range())){
		if(l.get(i) > m){
			a.put(l.get(i));
		}else{
			(b.put(l.get(i)));
		}
	}
	newList := a.append(m,b);

	# will not terminate trololololo
	quicksort(newList);
}
return 0;