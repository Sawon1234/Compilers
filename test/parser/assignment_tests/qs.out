fun quicksort < > (l : List < > ) : List < > { Integer < >  m := l . first < > ( ); l . pop < > ( ); List < > a := newList < > ( ) ; List < > b := newList < > ( ); for( i in 1 .. ( l . range < > ( ) ) ){ if( l . get < > ( i ) > m ){ a . put < > ( l . get < > ( i ) ) ; } else { ( b . put < > ( l . get < > ( i ) ) ) ; } } newList := a . flatten < > ( m , b ) ; quicksort < > ( newList ) ; } return 0 ;